### PR TITLE
Add logic to allow variants/* fields

### LINKED ISSUE #32: Add support for variants/{FIELD} in combine sample genotypes pipeline
cc @bstlaure

### PR DESCRIPTION
**Pull request by hardingnj**
Assumes field belongs to calldata.

Logic to add a dimension to 1D arrays found in either calldata or variants

Resolves #32.

### CODE REVIEW

File: scripts/combine_zarr.py [hunk 1]
```diff
@@ -112,13 +119,14 @@ def setup_output(output_path, seqid, field, example_arr, samples, cname, clevel,
     log('Setting up output at {!r} ...'.format(output_path))
     callset = zarr.open_group(output_path, mode='a')
     seq_group = callset.require_group(seqid)
-    calldata_group = seq_group.require_group('calldata')
+    field_root, field_id = field.split("/")
```
Comment (hardingnj) [inline scripts/combine_zarr.py:122]: Should I default to `calldata` here, irrespective of whether the original comes from `variants` or not?

**Comment by hardingnj** [comment]
nudge @malariagen/vector-core on this- thanks!

**Comment by alimanfoo** [comment]
Sorry for delay, taking a look now...

File: scripts/combine_zarr.py (hunk 1 above)
Comment (alimanfoo) [reply scripts/combine_zarr.py:122]: Yes I think always use `calldata` regardless of whether original comes from `calldata` or `variants`.

File: scripts/combine_zarr.py [hunk 2]
```diff
@@ -128,8 +136,12 @@ def setup_input(samples, input_pattern, seqid, field):
     input_paths = [input_pattern.format(sample=s) for s in samples]
     input_stores = [zarr.ZipStore(ip, mode='r') for ip in input_paths]
     input_roots = [zarr.group(store) for store in input_stores]
-    input_arrays = [root[s][seqid]['calldata'][field] for root, s in zip(input_roots, samples)]
+    input_arrays = [root[s][seqid][field] for root, s in zip(input_roots, samples)]
     input_arrays = [da.from_array(a, chunks=a.chunks) for a in input_arrays]
+
+    # here we add a dim to allow the hstack to work. must share the shape (X, 1, )
+    input_arrays = [a.reshape((-1, 1)) if a.ndim < 2 else a for a in input_arrays]
```
Comment (alimanfoo) [inline scripts/combine_zarr.py]:
```suggestion
    input_arrays = [a[:, None] if a.ndim == 1 else a for a in input_arrays]
```

Has the same effect but possibly slightly clearer that a new dimension is being inserted.

**Comment by alimanfoo** [comment]
Hi @hardingnj, just to say I think this is good to go, only a [couple of comments](https://github.com/malariagen/vector-tools/pull/37#pullrequestreview-252877306) outstanding, feel free to merge when addressed.

**Pull request merged**